csi/pkg/storage/modelregistry_provider.go

```diff
@@ -110,19 +105,84 @@ func (p *ModelRegistryProvider) DownloadModel(modelDir string, modelName string,
 	return provider.DownloadModel(modelDir, "", *modelArtifact.Uri)
 }
 
-func extractProtocol(storageURI string) (kserve.Protocol, error) {
+// Possible URIs:
+// (1) model-registry://{modelName}
+// (2) model-registry://{modelName}/{modelVersion}
+// (3) model-registry://{modelRegistryUrl}/{modelName}
+// (4) model-registry://{modelRegistryUrl}/{modelName}/{modelVersion}
+func (p *ModelRegistryProvider) parseModelVersion(storageUri string) (string, *string, error) {
+	var versionName *string
+
+	// Parse the URI to retrieve the needed information to query model registry (modelArtifact)
+	mrUri := strings.TrimPrefix(storageUri, string(constants.MR))
+
+	tokens := strings.SplitN(mrUri, "/", 3)
+
+	if len(tokens) == 0 || len(tokens) > 3 {
+		return "", nil, ErrInvalidMRURI
+	}
+
+	// Check if the first token is the host and remove it so that we reduce cases (3) and (4) to (1) and (2)
+	if len(tokens) >= 2 && p.Client.GetConfig().Host == tokens[0] {
+		tokens = tokens[1:]
+	}
+
```

Comment:
> My main concern with this assumption is that, if there is a model registry running but for any reason it is not accessible/reachable we are going to make the wrong assumption that it is a registeredModel name , right? And the error would be misleading to the user as it will find in the logs.

> We can improve this message telling the user that it failed to reach url from uri and it's going to use fallback url

Here (line 129) you have done all the storageURI parsing to determine in which case you are.
To me, add a Log info here that shows  ~circa

```
Parsed storageUri=... as: modelRegistryUrl=... modelName=... modelVersion=...
```

This way, we're being explicit on how we interpret what we have received based on the documentation provided. wdyt?

Reply:
I agree with you @tarilabs @Al-Pragliola , I think that providing a more meaningful and expecit log message would be enough for this use case!

Reply:
added
```golang
	log.Printf("Parsed storageUri=%s as: modelRegistryUrl=%s, registeredModelName=%s, versionName=%v",
		storageUri,
		p.Client.GetConfig().Host,
		registeredModelName,
		versionName,
	)
```